packages/webdriver-utils/src/providers/genericProvider.js

```diff
@@ -0,0 +1,127 @@
+import tmp from 'tmp';
+import fs from 'fs/promises';
+
+import CommonMetaDataResolver from '../metadata/commonMetaDataResolver.js';
+import Tile from '../util/tile.js';
+import Driver from '../driver.js';
+
+const log = utils.logger('webdriver-utils:genericProvider');
+// TODO: Need to pass parameter from sdk and catch in cli
+const CLIENT_INFO = 'local-poc-poa';
+const ENV_INFO = 'staging-poc-poa';
+
+export default class GenericProvider {
+  constructor(
+    sessionId,
+    commandExecutorUrl,
+    capabilities,
+    sessionCapabilites
+  ) {
+    this.sessionId = sessionId;
+    this.commandExecutorUrl = commandExecutorUrl;
+    this.capabilities = capabilities;
+    this.sessionCapabilites = sessionCapabilites;
+    this.driver = null;
+    this.commonMetaData = null;
+    this.debugUrl = null;
+  }
+
+  async createDriver() {
+    this.driver = new Driver(this.sessionId, this.commandExecutorUrl);
+    const caps = await this.driver.getCapabilites();
+    this.commonMetaData = await CommonMetaDataResolver.resolve(this.driver, caps, this.capabilities);
+  }
+
+  static supports(_commandExecutorUrl) {
+    return true;
+  }
+
+  async screenshot(name) {
+    let fullscreen = false;
+
+    const tag = await this.getTag();
+    const tiles = await this.getTiles(fullscreen);
+    await this.setDebugUrl();
+
+    log.debug(`${name} : Tag ${JSON.stringify(tag)}`);
+    log.debug(`${name} : Tiles ${JSON.stringify(tiles)}`);
+    log.debug(`${name} : Debug url ${this.debugUrl}`);
+    return await utils.postComparison({
+      name,
+      tag,
+      tiles,
+      // TODO: Fetch this one for bs automate, check appium sdk
+      externalDebugUrl: this.debugUrl,
+      environmentInfo: ENV_INFO,
+      clientInfo: CLIENT_INFO
+    });
+  }
+
+  async getTiles(fullscreen) {
+    if (!this.driver) throw new Error('Driver is null, please initialize driver with createDriver().');
+    const base64content = await this.driver.takeScreenshot();
+    const path = await this.writeTempImage(base64content);
+    return [
+      new Tile({
+        filepath: path,
+        // TODO: Need to add method to fetch these attr
+        statusBarHeight: 0,
+        navBarHeight: 0,
+        headerHeight: 0,
+        footerHeight: 0,
+        fullscreen
+      })
+    ];
+  }
+
+  async getTag() {
+    if (!this.driver) throw new Error('Driver is null, please initialize driver with createDriver().');
+    const { width, height } = await this.commonMetaData.windowSize();
+    const orientation = this.commonMetaData.orientation();
+    return {
+      name: this.commonMetaData.deviceName() || 'unknown',
+      osName: this.commonMetaData.osName() || 'unknown',
+      osVersion: this.commonMetaData.osVersion(),
+      width,
+      height,
+      orientation: orientation,
+      browserName: this.commonMetaData.browserName() || 'unknown',
+      // TODO
+      browserVersion: 'unknown'
+    };
+  }
+
+  async writeTempImage(base64content) {
```

Comment:
avoid this, directly pass base64 content [ its same process, its faster and efficient ] + we have seen cases where uses dont have access to temp locations